soccer/gameplay/tactics/wall.py

```diff
@@ -0,0 +1,65 @@
+import composite_behavior
+import behavior
+import constants
+import robocup
+import main
+import enum
+import math
+import skills.move
+
+# This tactic builds a wall a certain distance from point A, blocking point B.
+class Wall(composite_behavior.CompositeBehavior):
+    class State(enum.Enum):
+        defense_wall = 1
+
+    def __init__(self,
+                 num_defenders = 3,                         # number of defenders we're making the wall with (default 3)
+                 curvature = 0,                             # curvature (in radians) of the wall 0 - pi/2
+                 mark_point = None,                         # what point we are defending against (default is ball)
+                 defender_point = robocup.Point(0, 0),      # what point we are defending (default is goal)
+                 defender_spacing = 3,                      # number of robot radii between the centers of the defenders in the wall
+                 dist_from_mark = 1):                       # distance from the mark point we want to build the wall
+        super().__init__(continuous=True)
+
+        self.number_of_defenders = num_defenders
+        self.curvature = curvature
+        self.mark_point = main.ball().pos if mark_point == None else mark_point
+        self.defense_point = defender_point
+        self.dist_from_mark = dist_from_mark
+        self.defender_spacing = defender_spacing
+
+        self.add_state(Wall.State.defense_wall, 
+                       behavior.Behavior.State.running)
+
+        self.add_transition(behavior.Behavior.State.start,
+                            Wall.State.defense_wall, lambda: True,
+                            "immediately")
+
+    def on_enter_defense_wall(self):
+        self.remove_all_subbehaviors()
+        midpoint = self.arc_midpoint()
+        angle = (math.atan2(midpoint.x - self.defense_point.x, midpoint.y - self.defense_point.y))
+        for i in range(self.number_of_defenders):
+            pt = None
+            if self.number_of_defenders % 2 != 0:
+                pt = self.point_on_arc(midpoint, angle, i - int(self.number_of_defenders / 2))
+            else:
+                pt = self.point_on_arc(midpoint, angle, i - self.number_of_defenders / 2 + .5)
```

Comment:
Is this supposed to be a float still? If you want to round up, you can use the `ceil` function in the math library to round up to the nearest int

Reply:
The float value allows for a symmetrical, 'balanced,' wall when we have an even number of defenders. When we have an odd number of defenders, the direct shot is blocked by a single defender. When we have an even number of defenders, we have two defenders block the direct shot together, and using the float value, the wall is symmetric against the shot (if that makes any sense). Perhaps this should be re-thought because it may allow for small gap in the wall for a direct shot. 
![image](https://user-images.githubusercontent.com/21163393/52593303-9c6c1d00-2e16-11e9-8d80-dc4a0dc51fcf.png) (odd number of defenders)
![image](https://user-images.githubusercontent.com/21163393/52593365-bd347280-2e16-11e9-8100-4f9f3b81ba36.png) (even number of defenders)

Reply:
Yeah, I think it's an easy change based on testing. Let's just keep the current behavior since it's the "more complicated" version.

If I'm reading the code right, you actually don't need this if statement either. The `i - self.number_of_defenders / 2 + .5` works in both cases, even or odd. I came up the below which is the equivalent if you distribute the `-1` out of the parentheses.

`defender_number = (0:num_robots - 1) - (num_robots - 1) / 2`